### PR TITLE
fix: move Auth error codes to top level

### PR DESCRIPTION
**Pull request by J0**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Moves the Auth error codes

internal context: https://supabase.slack.com/archives/C022071RB2L/p1726139029552159

### CODE REVIEW

File: apps/docs/spec/common-client-libs-sections.json [hunk 1]
```diff
@@ -402,6 +402,7 @@
         "slug": "auth-error-codes",
         "type": "markdown",
         "excludes": [
+          "reference_javascript_v2",
```
Comment (J0) [inline apps/docs/spec/common-client-libs-sections.json:405]: Opted not to exclude entirely just yet as [Python uses error codes](https://supabase.com/docs/reference/python/auth-error-codes) and the implementation might differ. Can remove after checking with library owner

File: apps/docs/content/guides/auth/debugging/error-codes.mdx [hunk 2]
```diff
@@ -11,7 +20,10 @@ Errors originating from the server API classed as `AuthApiError` always have a `
 
 <AuthErrorCodesTable />
```
Comment (J0) [inline apps/docs/content/guides/auth/debugging/error-codes.mdx:21]: We can probably move the status codes out similar to what storage does. Are we okay with classifying them as "Legacy Error Codes"?

Comment (charislam) [reply apps/docs/content/guides/auth/debugging/error-codes.mdx:21]: Are they actually legacy? Or are they always going to convey the same information (just that this information is too coarse-grained to be useful)? We could call them "HTTP status codes" and put them beneath the other error codes

File: apps/docs/content/guides/auth/debugging/error-codes.mdx [hunk 3]
```diff
@@ -1,3 +1,12 @@
+---
+id: 'auth-error-codes'
+title: 'Error Codes'
+description: 'Supabase Auth Error Codes'
+subtitle: 'Learn about the Auth error codes and how to resolve them'
+---
+
+## Auth Error Codes
+
 Supabase Auth can throw or return various errors when using the API. All errors originating from the `supabase.auth` namespace of the client library will be wrapped by the `AuthError` class.
```
Comment (charislam) [inline apps/docs/content/guides/auth/debugging/error-codes.mdx:10]: One problem with this brought up by Tyler (and why he abstracted the AuthErrorCodes table in the first place) is that the intro is too JavaScript-centric (the listed classes are all JavaScript-based, for example). We might need to have tabs for the different language details

e.g., this is what Tyler wrote for [Dart](https://docs-git-j0-moveerrorcodes-supabase.vercel.app/docs/reference/dart/auth-error-codes)

File: apps/docs/spec/common-client-libs-sections.json (hunk 1 above)
Comment (charislam) [reply apps/docs/spec/common-client-libs-sections.json:405]: See comment above, the error codes table should be shared, we just need to move the intro text into tabs I think

File: apps/docs/content/guides/auth/debugging/error-codes.mdx (hunk 2 above)
Comment (J0) [reply apps/docs/content/guides/auth/debugging/error-codes.mdx:21]: I think they convey the same information but the information is too coarse-grained. HTTP status codes sounds fine yup

**Review by charislam — APPROVED** [review]

**Pull request merged**